File: email-backup.java/src/main/java/demo/Names.java
package demo;

import java.io.File;
import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.Date;

final public class Names {

    private static final String DATE_PATTERN = "dd-MM-yyyy HH.mm.ss";

    public static String getCurrentFolderName() throws IOException {
        return new File(".").getCanonicalPath();
    }

    public static String getZipName(String folderName, Date date) {
        return getZipNamePrefix(folderName) + getZipNameSuffix(date);
    }

    static String getZipNamePrefix(String folderName) {
        int i = folderName.lastIndexOf(File.separator);
        return (i < 0) ? folderName : folderName.substring(i + 1);
    }

    static String getZipNameSuffix(Date date) {
        return " [" + new SimpleDateFormat(DATE_PATTERN).format(date) + "].zip";
    }
}

File: email-backup.kotlin/src/main/kotlin/demo/Names.kt
package demo

import java.io.File
import java.io.IOException
import java.text.SimpleDateFormat
import java.util.Date

public object Names {

    private val DATE_PATTERN = "dd-MM-yyyy HH.mm.ss"

    @Throws(IOException::class)
    public fun getCurrentFolderName(): String {
        return File(".").canonicalPath
    }

    public fun getZipName(folderName: String, date: Date): String {
        return getZipNamePrefix(folderName) + getZipNameSuffix(date)
    }

    fun getZipNamePrefix(folderName: String): String {
        val i = folderName.lastIndexOf(File.separator)
        return if ((i < 0)) folderName else folderName.substring(i + 1)
    }

    fun getZipNameSuffix(date: Date): String {
        return " [" + SimpleDateFormat(DATE_PATTERN).format(date) + "].zip"
    }
}
